src/EditorFeatures/Text/Extensions.TextBufferContainer.cs

```diff
@@ -100,18 +100,21 @@ public override SourceText CurrentText
             private void OnTextContentChanged(object sender, TextContentChangedEventArgs args)
             {
                 var changed = this.EtextChanged;
-                if (changed != null && args.Changes.Count != 0)
+                if (changed == null)
```

Comment:
I would suggest a comment indicating that it's important to fire the event even if there are no changes so that consumers can observe the new snapshot.

Otherwise, someone is likely to re-add the optimization at some point in the future.

Reply:
will do